File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsSearchHistoryMapper.java
package top.charjin.shoppingserver.mapper;

import top.charjin.shoppingserver.entity.OsSearchHistory;

public interface OsSearchHistoryMapper {
    int deleteByPrimaryKey(Integer id);

    int insert(OsSearchHistory record);

    int insertSelective(OsSearchHistory record);

    OsSearchHistory selectByPrimaryKey(Integer id);

    int updateByPrimaryKeySelective(OsSearchHistory record);

    int updateByPrimaryKey(OsSearchHistory record);
}

File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsSearchHistoryMapper.kt
package top.charjin.shoppingserver.mapper

import top.charjin.shoppingserver.entity.OsSearchHistory

interface OsSearchHistoryMapper {
    fun deleteByPrimaryKey(id: Int?): Int

    fun insert(record: OsSearchHistory): Int

    fun insertSelective(record: OsSearchHistory): Int

    fun selectByPrimaryKey(id: Int?): OsSearchHistory

    fun updateByPrimaryKeySelective(record: OsSearchHistory): Int

    fun updateByPrimaryKey(record: OsSearchHistory): Int
}